frontend/pages/register.vue

```diff
@@ -0,0 +1,176 @@
+<script setup>
+definePageMeta({
+  middleware: 'guest',
+});
+useSeoMeta({
+  title: 'Créer un compte - PopnBed',
+  meta: [
+    {
+      name: 'description',
+      content: 'Créez un compte sur PopnBed pour réserver des hébergements inspirés de films.',
+    },
+  ],
+});
+const authStore = useAuthStore();
+const router = useRouter();
+
+const form = reactive({
+  email: ref(''),
+  firstName: ref(''),
+  lastName: ref(''),
+  password: ref(''),
+  confirmPassword: ref('')
+});
+
+const errors = reactive({});
+const isSubmitting = ref(false);
+
+const resetErrors = () => {
+  const keys = Object.keys(errors);
+  keys.forEach(key => {
+    errors[key] = undefined;
+  });
+};
+
+const register = async () => {
+  resetErrors();
+
+  let isValid = true;
+
+  if (!form.email) {
+    errors.email = "L'email est requis.";
+    isValid = false;
+  } else if (!/\S+@\S+\.\S+/.test(form.email)) {
+    errors.email = "L'email n'est pas valide.";
+    isValid = false;
+  }
+
+  if (!form.firstName) {
+    errors.firstName = "Le prénom est requis.";
+    isValid = false;
+  }
+
+  if (!form.lastName) {
+    errors.lastName = "Le nom est requis.";
+    isValid = false;
+  }
+
+  if (!form.password) {
+    errors.password = "Le mot de passe est requis.";
+    isValid = false;
+  }
+
+  if (form.password !== form.confirmPassword) {
+    errors.confirmPassword = "Les mots de passe ne correspondent pas";
+    isValid = false;
+  }
+
+  if (!isValid) return;
+
+  isSubmitting.value = true;
+
+  try {
+    const result = await authStore.register({
+      email: form.email,
+      firstName: form.firstName,
+      lastName: form.lastName,
+      username: form.email,
+      password: form.password
+    });
+
+    if (result.success) {
+      router.push('/');
+    } else {
+      errors.email = result.message;
+    }
+
+  } catch (error) {
+    if (error.response?.data?.errors) {
+      Object.assign(errors, error.response.data.errors);
+    }
+
+    if (error.response?.data?.message) {
+      errors.email = error.response.data.message;
+    }
+
+  } finally {
+    isSubmitting.value = false;
+  }
+};
+</script>
+
+<template>
+  <main class="w-full h-full flex justify-center pt-20">
+    <UGridBackgroundPattern class="absolute top-0"/>
+    <section class="flex flex-col items-center space-y-6 relative">
+      <div class="text-center space-y-3">
+        <h1 class="text-primary font-semibold text-3xl">Créer un compte</h1>
+        <p class="text-tertiary font-normal">Veuillez entrer vos informations personnelles</p>
+      </div>
+      <form class="w-96" @submit.prevent="register">
+        <div class="space-y-4">
+          <UInput
+              v-model="form.email"
+              type="email"
+              name="email"
+              placeholder="Email"
+              label="Email"
+              :error="errors.email"
```

Comment:
Ici le `:error` n'existe pas, il faut utiliser le `hintText` et la prop `destructive` pour qu'il soit afficher en rouge.
```suggestion
              :hintText="errors.email"
              :destructive="errors.email"
```